integration-tests/src/test/java/oracle/kubernetes/operator/ITServerDiscovery.java

```diff
@@ -0,0 +1,541 @@
+
+  private static boolean QUICKTEST;
+  private static boolean SMOKETEST;
+  private static boolean JENKINS;
+  private static boolean INGRESSPERDOMAIN = true;
+
+  // Set QUICKTEST env var to true to run a small subset of tests.
+  // Set SMOKETEST env var to true to run an even smaller subset
+  // of tests, plus leave domain1 up and running when the test completes.
+  // set INGRESSPERDOMAIN to false to create LB's ingress by kubectl yaml file
+  static {
+    QUICKTEST =
+        System.getenv("QUICKTEST") != null && System.getenv("QUICKTEST").equalsIgnoreCase("true");
+    SMOKETEST =
+        System.getenv("SMOKETEST") != null && System.getenv("SMOKETEST").equalsIgnoreCase("true");
+    if (SMOKETEST) QUICKTEST = true;
+    if (System.getenv("JENKINS") != null) {
+      JENKINS = new Boolean(System.getenv("JENKINS")).booleanValue();
+    }
+    if (System.getenv("INGRESSPERDOMAIN") != null) {
+      INGRESSPERDOMAIN = new Boolean(System.getenv("INGRESSPERDOMAIN")).booleanValue();
+    }
+  }
+
+  /**
+   * This method gets called only once before any of the test methods are executed. It does the
+   * initialization of the integration test properties defined in OperatorIT.properties and setting
+   * the resultRoot, pvRoot and projectRoot attributes.
+   *
+   * <p>It also create operator and verify its deployed successfully. Create domain and verify
+   * domain is created.
+   *
+   * @throws Exception
+   */
+  @BeforeClass
+  public static void staticPrepare() throws Exception {
+    Assume.assumeFalse(QUICKTEST);
+    // initialize test properties and create the directories
+    initialize(appPropsFile);
+
+    // create operator1
+    if (operator == null) {
+      logger.info("Creating Operator & waiting for the script to complete execution");
+      operator = TestUtils.createOperator(operatorYmlFile);
+    }
+
+    // create domain
+    if (domain == null) {
+      logger.info("Creating WLS Domain & waiting for the script to complete execution");
+      domain = TestUtils.createDomain(domainonpvwlstFile);
+      domain.verifyDomainCreated();
+    }
+
+    domainYaml =
+        BaseTest.getUserProjectsDir()
+            + "/weblogic-domains/"
+            + domain.getDomainUid()
+            + "/domain.yaml";
+
+    // create test domain yaml file
+    testDomainYamlFile = BaseTest.getResultDir() + "/" + testYamlFileName;
+
+    Path sourceFile = Paths.get(domainYaml);
+    Path targetFile = Paths.get(testDomainYamlFile);
+
+    Files.copy(sourceFile, targetFile, StandardCopyOption.REPLACE_EXISTING);
+
+    String content = new String(Files.readAllBytes(targetFile), StandardCharsets.UTF_8);
+    content = content.replaceAll("replicas: 2", "replicas: 3");
+    Files.write(targetFile, content.getBytes(StandardCharsets.UTF_8));
+  }
+
+  /**
+   * Releases k8s cluster lease, archives result, pv directories
+   *
+   * @throws Exception
+   */
+  @AfterClass
+  public static void staticUnPrepare() throws Exception {
```

Comment:
Add  Assume.assumeFalse(QUICKTEST); at the beginning

Reply:
Ok. Will do

I dind't add it there because of the comments from the link below :(. But in our case, you are right, we should add it to staticUnPrepare() since we call cleanup() there.

https://stackoverflow.com/questions/1689242/conditionally-ignoring-tests-in-junit-4

Reply:
Just for record purpose, we can add Assume.assumeFalse(QUICKTEST); either in staticPrepare or staticUnprepare, but not in BOTH. both Vanaja and I verified it